data/moves.ts

```diff
@@ -13396,8 +13415,9 @@ export const Moves: {[moveid: string]: MoveData} = {
 		priority: 0,
 		flags: {protect: 1, reflectable: 1, heal: 1},
 		onHit(target, source) {
-			if (!target.cureStatus()) return false;
-			this.heal(Math.ceil(source.maxhp * 0.5), source);
+			if (!target.cureStatus()) return this.NOT_FAIL;
+			const success = !!this.heal(Math.ceil(source.maxhp * 0.5), source);
+			return success ? success : this.NOT_FAIL;
```

Comment:
``success || this.NOT_FAIL`` won't work because ``true || ''`` is true and ``false || ''`` is false.

Hardcoding in this.NOT_FAILs everywhere really isn't the correct solution anyway; the correct thing to do would be move the healing checks at some point after where Stomping Tantrum should fail, but I think that requires a massive overhaul of how event failure works.

Reply:
Sorry, that was a typo on my part, I meant to use `&&` instead of `||` of course.

Reply:
I don't think that works either unless I'm misunderstanding. ``true && ''`` is ``''``, and ``false && ''`` is ``false``.

Reply:
No, it was my fault for believing you when you said `false || ''` is `false`. It's not.

Reply:
Bah, I feel like a dunce. >.<